tests/Microsoft.Data.Analysis.Tests/DataFrameTests.cs

```diff
@@ -1881,5 +1881,15 @@ public void TestAppendRow()
             Assert.Equal(5, df.Columns[0].NullCount);
             Assert.Equal(6, df.Columns[1].NullCount);
         }
+
+        [Fact]
+        public void TestAppendEmptyValue()
+        {
+            DataFrame df = MakeDataFrame<int, bool>(10);
+            df.Append(new List<object> { "", true });
+            Assert.Equal(11, df.RowCount);
+            Assert.Equal(2, df.Columns[0].NullCount);
```

Comment:
An empty string is not `null`. It should be treated separately.

Reply:
I agree that an empty string should not be `null` in a `StringDataFrameColumn`, but what about an empty string in a primitive column? Our options are either to treat it as `null` or throw. I didn't think we should throw because many csv files seem to have empty strings and we still want to be able to read them. With `null`s, users could set them to whatever value they want.

Reply:
I agree it should be `null` for non-string columns like `int`, etc.